### DISCUSSION TITLE
If cuts aren't mutable how does one change the text of a cut?

### DISCUSSION THREAD

**Discussion by RuABraun** [lhotse-speech/lhotse#1179] [Q&A] [answered]
Hi,

I have a cut and want to modify its text. Assigning `cut.supervisions[0].text='bla'` does not work as expected, since it's not mutable. I don't see an explanation for what to do if one needs to modify cuts/cutsets though, what can I do?

**Comment by desh2608** [comment, marked as answer]
I think the following should work:
```python
from lhotse.utils import fastcopy

new_cut = fastcopy(cut, supervisions=[fastcopy(cut.supervisions[0], text="bla")])
```

Also check out: `CutSet.map_supervisions()`, `CutSet.transform_text()`.

**Reply by RuABraun** [reply]
thank you! :)

for larger changes I guess it's unavoidable to loop over and create new SupervisionSet and RecordingSets ?

**Reply by desh2608** [reply]
yeah, might be best to create a new supervision manifest with the next text. BTW, `Cut` supports custom fields, so you can add a new text to an existing cut under a new field name.
